masq/src/commands/exit_location_command.rs

```diff
@@ -0,0 +1,533 @@
+        let stderr = context.stderr_arc();
+        let stdout = context.stdout_arc();
+        assert_eq!(
+            stdout.lock().unwrap().get_string(),
+            "All requested countries are missing in Database: \"CZ, SK, IN\"\n"
+        );
+        assert_eq!(
+            stderr.lock().unwrap().get_string(),
+            "code: 9223372036854775817\nmessage: \"CZ, SK, IN\"\n".to_string()
+        );
+        assert_eq!(
+            result,
+            Err(CommandError::Payload(
+                9223372036854775817,
+                "\"CZ, SK, IN\"".to_string()
+            ))
+        );
+    }
+
+    #[test]
+    fn testing_exit_location_general_error() {
+        let factory = CommandFactoryReal::new();
+        let mut context = CommandContextMock::new().transact_result(Err(
+            ContextError::PayloadError(EXIT_COUNTRY_ERROR, "".to_string()),
+        ));
+        let subject = factory.make(&["exit-location".to_string()]).unwrap();
+
+        let result = subject.execute(&mut context);
+        let stderr = context.stderr_arc();
+        let stdout = context.stdout_arc();
+        assert!(stdout.lock().unwrap().get_string().is_empty());
+        assert_eq!(
+            stderr.lock().unwrap().get_string(),
+            "Error: Something went wrong!\n".to_string()
+        );
+        assert_eq!(result, Err(Payload(9223372036854775816, "".to_string())));
+    }
+
+    #[test]
+    fn testing_handler_for_exit_location_responose() {
+        let message_body = Ok(UiSetExitLocationResponse {
+            fallback_routing: false,
+            exit_locations: vec![
+                ExitLocation {
+                    country_codes: vec!["CZ".to_string()],
+                    priority: 1,
+                },
+                ExitLocation {
+                    country_codes: vec!["FR".to_string()],
+                    priority: 2,
+                },
+            ],
+            exit_countries: Some(vec!["FR".to_string()]),
+            missing_countries: vec!["CZ".to_string()],
+        }
+        .tmb(1234));
+
+        let factory = CommandFactoryReal::new();
+        let mut context = CommandContextMock::new().transact_result(message_body);
+        let subject = factory.make(&["exit-location".to_string()]).unwrap();
+
+        let result = subject.execute(&mut context);
+        let stderr = context.stderr_arc();
+        let stdout = context.stdout_arc();
+        assert_eq!(stdout.lock().unwrap().get_string(), "Countries available for exit-location: [\"FR\"]\nFallback Routing NOT set.\nFollowing countries are missing in Database: [\"CZ\"]\nExit location set: Country Codes: [\"CZ\"] - Priority: 1; Country Codes: [\"FR\"] - Priority: 2; \n".to_string());
+        assert_eq!(
+            stderr.lock().unwrap().get_string(),
+            "code: 9223372036854775817\nmessage: [\"CZ\"]\n".to_string()
+        );
+        assert_eq!(result, Ok(()));
+    }
+
+    #[test]
+    fn can_deserialize_ui_set_exit_location() {
```

Comment:
Jsem trochu skeptycky k tomuto druhu testu psanem pro ty masq prikazy, protoze ty ostatni testy tady implicitne dokazuji, ze serializace a deserializace dat funguje spravne. Tudiz neni potreba mit jeden extra, ktery bude tvrdit, ze testuje prave a pouze toto. Za me, pokud ho cely smazes, vubec nic se nestane. Melo by to jen benefity, protoze bychom se nemuseli do budoucna o ten test starat.